stubs/pluggy/pluggy/_tracing.pyi

```diff
@@ -0,0 +1,20 @@
+from collections.abc import Callable
+from typing import Any
+from typing_extensions import TypeAlias
+
+_Writer: TypeAlias = Callable[[str], object]
+_Processor: TypeAlias = Callable[[tuple[str, ...], tuple[Any, ...]], None]
+
+class TagTracer:
+    indent: int
+    def __init__(self) -> None: ...
+    def get(self, name: str) -> TagTracerSub: ...
+    def setwriter(self, writer: _Writer) -> None: ...
```

Comment:
pytest does `setwriter(None)` here: https://github.com/pytest-dev/pytest/blob/1de0923e83e3258709559d86d142adaab65a42d8/src/_pytest/helpconfig.py#L126

And that makes sense to me, since `setwriter` just sets the private `_writer` variable: https://github.com/pytest-dev/pluggy/blob/e507275d33b03b7e7161b2d8c9b184939973bc91/src/pluggy/_tracing.py#L53

And the `_writer` variable is typed as `_Writer | None` in the pluggy source code: https://github.com/pytest-dev/pluggy/blob/e507275d33b03b7e7161b2d8c9b184939973bc91/src/pluggy/_tracing.py#L19

So I think the upstream annotation for this method is wrong, and it should be:

```suggestion
    def setwriter(self, writer: _Writer | None) -> None: ...
```

(This gets rid of another hit that mypy_primer surfaced)

Reply:
@AlexWaygood good catch, I took the liberty to propose this change into upstream as well (https://github.com/pytest-dev/pluggy/pull/423)

Reply:
> @AlexWaygood good catch, I took the liberty to propose this change into upstream as well ([pytest-dev/pluggy#423](https://github.com/pytest-dev/pluggy/pull/423))

Nice! https://github.com/python/typeshed/pull/10473#discussion_r1268585883 could also probably be contributed upstream, if you feel like it!